client/app/bundles/course/achievement/reducers.ts

```diff
@@ -0,0 +1,70 @@
+import { produce } from 'immer';
+import {
+  AchievementCourseUserEntity,
+  AchievementEntity,
+  AchievementMiniEntity,
+} from 'types/course/achievements';
+import {
+  createEntityStore,
+  removeFromStore,
+  saveEntityToStore,
+  saveListToStore,
+} from 'utilities/store';
+import * as types from './types';
+
+const initialState: types.AchievementsState = {
+  achievements: createEntityStore(),
```

Comment:
As discussed, this redux framework will be good for caching but may add another level of complexity for developers. We will keep it as of now since it wont be difficult to simplify the redux store further down the road.